src/test/java/com/rj/game/user/create/UserIdExtractorTest.java

```diff
@@ -0,0 +1,37 @@
+package com.rj.game.user.create;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import java.net.URI;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.UUID;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class UserIdExtractorTest {
+
+    private UserIdExtractor userIdExtractor;
+
+    @BeforeEach
+    public void setUp() {
+        userIdExtractor = new UserIdExtractor();
+    }
+
+    @Test
+    public void testExtractValidUserId() {
```

Comment:
poczytaj też o https://javastart.pl/baza-wiedzy/testowanie-jednostkowe/given-when-then-w-testach i zastosuj to w swoich testach